code/game/atoms.dm

```diff
@@ -349,18 +370,117 @@
 	var/new_blood_dna = L.get_blood_dna_list()
 	if(!new_blood_dna)
 		return FALSE
-	var/old_length = blood_DNA_length()
-	add_blood_DNA(new_blood_dna)
-	if(blood_DNA_length() == old_length)
+	LAZYINITLIST(blood_DNA)	//if our list of DNA doesn't exist yet, initialise it.
+	var/old_length = blood_DNA.len
+	blood_DNA |= new_blood_dna
+	if(blood_DNA.len == old_length)
 		return FALSE
 	return TRUE
 
+//to add blood dna info to the object's blood_DNA list
+/atom/proc/transfer_blood_dna(list/blood_dna, list/datum/disease/diseases)
+	LAZYINITLIST(blood_DNA)
+	var/old_length = blood_DNA.len
+	blood_DNA |= blood_dna
+	if(blood_DNA.len > old_length)
+		return TRUE
+		//some new blood DNA was added
+
+
 //to add blood from a mob onto something, and transfer their dna info
 /atom/proc/add_mob_blood(mob/living/M)
 	var/list/blood_dna = M.get_blood_dna_list()
 	if(!blood_dna)
 		return FALSE
-	return add_blood_DNA(blood_dna)
+	return add_blood_DNA(blood_dna, M.diseases)
+
+//to add blood onto something, with blood dna info to include.
+/atom/proc/add_blood_DNA(list/blood_dna, list/datum/disease/diseases)
+	return FALSE
+
+/obj/add_blood_DNA(list/blood_dna, list/datum/disease/diseases)
+	return transfer_blood_dna(blood_dna, diseases)
+
+/obj/item/add_blood_DNA(list/blood_dna, list/datum/disease/diseases)
+	if(!..())

```

Comment:
> Unresolving since this is a much cleaner method of handling input from parent procs
> EDIT: since github doesnt display the original comment in the review, here's a copy-paste
> 
> > Do this instead:
> > ```
> > 	. = ..()
> > 	if(!.)
> > ```
> > 
> > 
> > This way you can also pass down the parent return value to possible subtypes' own definitions of this function. Little coding standard.

Still needs addressing.

Reply:
it was attempted, and found to not be working.

Reply:
How exactly does it not work?

Reply:
> the suggestion of
> 
> . = ..()
> 	if(!.)
> in atom.dm also did not work. it would return the TRUE as a FALSE and then fail the entire proc. because byond.

Reply:
Was it runtiming? Did you follow Ghommie's other two comments in this proc?

Reply:
It wasn't run-timing, because it encountered the if false return, so it didn't do anything. Wasn't adding overlays as a result.

The other two comments required it to work at all. I even added debug text to see what the fuck was going on, and it would say shit like 'Atom level returned true' 'item level returned false' I vocalized a great majority of my frustration in the discord.